### PR TITLE
feat: enhance call-ai with automatic Vibes authentication

### PR DESCRIPTION
**Pull request by jchris**
## Summary

Implement direct enhancement of the call-ai package to automatically inject X-VIBES-Token headers when available, providing transparent authentication for Vibe apps running on vibesdiy.net.

This approach avoids the circular import issues we encountered with the wrapper strategy by enhancing call-ai directly, following the same successful pattern as enhanced useFireproof.

## Key Features

- **Automatic auth token injection** from localStorage (`auth_token`)
- **Preserves caller-provided headers** - respects existing X-VIBES-Token if provided  
- **Graceful degradation** - works seamlessly in non-browser environments
- **No circular imports** - enhancement built directly into call-ai package
- **Comprehensive test coverage** - 10 test scenarios covering all edge cases

### Enhanced callAi Function
- Added `enhanceWithVibesAuth()` helper that wraps options with auth headers
- Auth token retrieved from `localStorage.getItem('auth_token')`  
- Enhancement applied at function entry, affecting all code paths (streaming, non-streaming, buffered)

### Approaches Considered

1. **Enhanced wrapper approach** ❌ - Caused circular imports when import maps redirect call-ai → use-vibes
2. **Escape hatches with /original paths** ❌ - Complex and fragile import map management  
3. **Dynamic imports** ❌ - Runtime complexity and potential race conditions
4. **Direct enhancement** ✅ - Clean, simple, follows proven useFireproof pattern

### Test Coverage

Created `call-ai/tests/unit/vibes-auth.test.ts` with comprehensive scenarios:
- Auth token injection when available
- Preservation of caller-provided tokens
- Graceful handling when no token available  
- Header preservation with mixed headers
- Error handling for localStorage access issues
- Non-browser environment compatibility
- Streaming and non-streaming request paths
- Empty/falsy token handling

## Benefits

- **Transparent authentication** for all Vibe apps automatically
- **Zero breaking changes** - fully backward compatible
- **Simple architecture** - no complex wrapper or import management
- **Future-proof** - easy to extend with additional Vibes features
- **Proven pattern** - follows successful enhanced useFireproof approach

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: call-ai/pkg/api.ts [hunk 1]
```diff
@@ -22,6 +22,53 @@ import { createStreamingGenerator } from "./streaming.js";
...
+
+  // Add the auth token to headers
+  return {
+    ...options,
+    headers: {
+      ...(options.headers || {}),
+      "X-VIBES-Token": authToken,
+    },
+  };
+}
```
[the remark this answers is not in the document]
Comment (jchris) [reply call-ai/pkg/api.ts]: @CharlieHelps yes please
